Automattic/wp-calypso
# Checkout: remove post-checkout polling for site for create_new_blog

#### Background

The transactions endpoint supports a property on the shopping cart object called `create_new_blog`. If set, the endpoint will create a new site before the transaction is processed. However, even though [we set this property on the frontend](https://github.com/Automattic/wp-calypso/blob/2c707eaab587ae48cd342fe76d37546eed2a68ef/client/my-sites/checkout/composite-checkout/lib/translate-cart.ts#L89) for any transaction where there is no currently selected site (which should be domain-only checkout and Jetpack siteless checkout, although [the latter may have a special case](https://github.com/Automattic/wp-calypso/blob/2c707eaab587ae48cd342fe76d37546eed2a68ef/client/my-sites/checkout/composite-checkout/components/checkout-main.tsx#L200)), it doesn't appear to be ever used on the backend to create a new site (see https://github.com/Automattic/payments-shilling/issues/759#issuecomment-1306365375). This is because [logged-out and siteless flows](https://github.com/Automattic/wp-calypso/blob/17f26290ffabb9a9d6437d5166a0e8f5265cd70d/client/my-sites/checkout/composite-checkout/components/checkout-main.tsx#L139) that currently go through checkout (except for Jetpack siteless checkout which understandably does not need to create a site) [create the site before the transaction is submitted](https://github.com/Automattic/wp-calypso/blob/17f26290ffabb9a9d6437d5166a0e8f5265cd70d/client/my-sites/checkout/composite-checkout/lib/submit-wpcom-transaction.ts#L32-L34).

While it may not yet be safe to remove the `create_new_blog` property entirely (since it may be used on the backend cart to signal that a site _will_ be created), we can probably remove some of the machinery that uses it.

#### Proposed Changes

In this PR we remove code that causes some post-checkout flows (for non-redirect payment methods which have `create_new_blog` and a domain in the purchase) to poll the sites endpoint after the transactions endpoint returns a successful response. This code was probably intended to prevent issues where the post-checkout page would try to display information about the newly created site and fail, but I think this is unlikely ever to happen since post-checkout for new sites already [causes a hard-redirect](https://github.com/Automattic/wp-calypso/blob/17f26290ffabb9a9d6437d5166a0e8f5265cd70d/client/my-sites/checkout/composite-checkout/hooks/use-create-payment-complete-callback.tsx#L195-L199) which will probably reload the sites list.

The code removed by this PR was added in https://github.com/Automattic/wp-calypso/pull/42723 which in turn imported the behavior from the old version of checkout where it was added originally in https://github.com/Automattic/wp-calypso/pull/10936. That PR was creating a flow for domain-only checkout which should not be affected by this change (although I think that flow does create a hidden site in the background so it's wise to make sure that PR's test instructions still work).

Fixes https://github.com/Automattic/wp-calypso/issues/69833 where purchasing a gift subscription for another site would cause infinite polling to occur.

Note: this also removes the `getDomainNameFromReceiptOrCart()` function which is no longer used anywhere.

#### Testing Instructions

First we can follow the test instructions for the PR that added this code to the rebuilt checkout:

- Create a new site and select a paid domain and a paid plan during the site creation process.
- Complete checkout during the site creation process.
- Verify that the post-checkout flow refers to the site just created.

Second we can follow the instructions for the PR that originally added the code to make sure that domain-only checkout is unaffected:

- Visit `/start/domain/domain-only` and complete the flow to purchase a new domain without a site.
- Verify that when checkout completes, you are redirected to a post-checkout page for that domain.
